narwhals/_arrow/dataframe.py

```diff
@@ -324,3 +329,55 @@ def collect(self) -> ArrowDataFrame:
 
     def clone(self) -> Self:
         raise NotImplementedError("clone is not yet supported on PyArrow tables")
+
+    def is_empty(self: Self) -> bool:
+        return self.shape[0] == 0
+
+    def item(self: Self, row: int | None = None, column: int | str | None = None) -> Any:
+        if row is None and column is None:
+            if self.shape != (1, 1):
+                msg = (
+                    "can only call `.item()` if the dataframe is of shape (1, 1),"
+                    " or if explicit row/col values are provided;"
+                    f" frame has shape {self.shape!r}"
+                )
+                raise ValueError(msg)
+            return self._native_dataframe[0][0].as_py()
+
+        elif row is None or column is None:
+            msg = "cannot call `.item()` with only one of `row` or `column`"
+            raise ValueError(msg)
+
+        _col = self.columns.index(column) if isinstance(column, str) else column
+        return self._native_dataframe[_col][row].as_py()
+
+    def iter_rows(
+        self,
+        *,
+        named: bool = False,
+        buffer_size: int = 512,  # noqa: ARG002
+    ) -> Iterator[tuple[Any, ...]] | Iterator[dict[str, Any]]:
+        """
+        NOTE:
+            The param ``buffer_size`` is only here for compatibility with the polars API
+            and has no effect on the output.
+        """
+        df = self._native_dataframe
+
+        num_rows = df.num_rows
+        columns = df.column_names
+
+        for row in range(num_rows):
+            if not named:
+                yield tuple(df[_col][row].as_py() for _col in columns)
+            else:
+                yield {_col: df[_col][row].as_py() for _col in columns}
```

Comment:
that's a lot of `.as_py` calls, should we be concern about performance here?

Reply:
hmm maybe...when I asked about iter_rows vs rows to polars devs, they said that doing it in buffers made a massive difference

if there's not a way to do that in pyarrow, maybe we can leave this one out for now?

Reply:
I was not able to find any reference on how to get an entire row performantly, especially as a python object.

> we can leave this one out for now?

What about adding a performance warning? After all, one should know that arrow has columnar data format, and we are asking for row iteration 🙈

Reply:
Ok I probably need to investigate further then. Let me leave `iter_rows` out of this PR